### ISSUE TITLE
feature: support JPMS

### ISSUE THREAD

**Issue by xenoterracide** [Netflix/dgs-framework#1192]
Please read our [contributor guide](https://github.com/Netflix/dgs-framework/blob/master/CONTRIBUTING.md) before
creating an issue. Also consider discussing your idea on
the [discussion forum](https://github.com/Netflix/dgs-framework/discussions) first.

## Describe the Feature Request

I'd like to be able to import dgs into an app using java modules. I searched the repo for a `module-info.java` or an `Automatic-Module-Name` Given the absence of the former it makes sense to do the latter.

## Describe Preferred Solution

You can add an `Automatic-Module-Name` to the manifest https://stackoverflow.com/a/46742802/206466 this is not backwards incompatible. A `module-info` would be nice, but it will  force people to java 9+

## Describe Alternatives

Without any support applications have to remove there's You simply can't import a module that doesn't have one of these if you're using it.

**Comment by paulbakker** [comment]
I agree, let's start with adding a module name, so that at least it becomes usable. 

Once we baseline on Spring Boot 3.0, which requires Java 17, it makes sense to add module-info as well. There isn't a concrete plan for that yet, but will likely be early 2023 (since Spring Boot 3 will be release end of this year).

**Comment by xenoterracide** [comment]
btw, any chance what's the potential itinerary on `Automatic-Module-Name`?

**Comment by xenoterracide** [comment]
2 year bump... ?

**Comment by paulbakker** [comment]
Implemented (finally 😅) in https://github.com/Netflix/dgs-framework/releases/tag/v10.1.0
